hack/local-up-cluster.sh

```diff
@@ -280,39 +280,39 @@ function detect_binary {
 
     case "$(uname -m)" in
       x86_64*)
-        host_arch=amd64
+        HOST_ARCH=amd64
         ;;
       i?86_64*)
-        host_arch=amd64
+        HOST_ARCH=amd64
         ;;
       amd64*)
-        host_arch=amd64
+        HOST_ARCH=amd64
         ;;
       aarch64*)
-        host_arch=arm64
+        HOST_ARCH=arm64
         ;;
       arm64*)
-        host_arch=arm64
+        HOST_ARCH=arm64
         ;;
       arm*)
-        host_arch=arm
+        HOST_ARCH=arm
         ;;
       i?86*)
-        host_arch=x86
+        HOST_ARCH=x86
         ;;
       s390x*)
-        host_arch=s390x
+        HOST_ARCH=s390x
         ;;
       ppc64le*)
-        host_arch=ppc64le
+        HOST_ARCH=ppc64le
         ;;
       *)
         echo "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x or ppc64le." >&2
         exit 1
         ;;
     esac
 
-   GO_OUT="${KUBE_ROOT}/_output/local/bin/${host_os}/${host_arch}"
+   GO_OUT="${KUBE_ROOT}/_output/local/bin/${HOST_OS}/${HOST_ARCH}"
```

Comment:
What's the point of this renaming? Even if HOST_ARCH and HOST_OS are initially set, this code would not be reached if host OS or arch are not detected.

Reply:
If you want to use host os and arch values you should consider returning them from this function instead of changing global variables.

Reply:
> What's the point of this renaming

The reason for setting HOST_ARCH and HOST_OS as global variables is that need to be referenced in the install_cni function

Reply:
If we want to pass variables using return values, how about the following?
```diff
+function detect_arch {
+    local host_arch
+
+    case "$(uname -m)" in
+      x86_64*)
+        host_arch=amd64
+        ;;
+      i?86_64*)
+        host_arch=amd64
+        ;;
+
+        ......
+
+    esac
+
+  echo -n "${host_arch}"
+}
+
+function detect_os {
+    local host_os
+
+    case "$(uname -s)" in
+      Darwin)
+        host_os=darwin
+        ;;
+
+      ......
+
+    esac
+
+  echo -n "${host_os}"
+}

function detect_binary {
+    host_arch=$(detect_arch)
+    host_os=$(detect_os)
-    case "$(uname -s)" in
-      Darwin)
-        host_os=darwin
-        ;;
-      Linux)
- 
-       ......
- 
-    esac
    GO_OUT="${KUBE_ROOT}/_output/local/bin/${host_os}/${host_arch}"
}

function install_cni {
+ host_arch=$(detect_arch)
+ cni_plugin_sha=CNI_PLUGINS_${host_arch^^}_SHA256SUM
+ cni_plugin_tarball="${CNI_PLUGINS_VERSION}/cni-plugins-linux-${host_arch}-${CNI_PLUGINS_VERSION}.tgz"
  ....
}
```

Reply:
This looks better to me, thanks.